### PR TITLE
Cart 912

### PR DESCRIPTION
**Pull request by SABollinger**

### CODE REVIEW

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1981]: Added a break, not necesary.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1767]: Fixed alignment

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c]: Fixed alignment.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c]: Fixed alignment.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1799]: Fixed alignment.

File: src/utils/self_test/INSTALL
Comment (SABollinger) [reply src/utils/self_test/INSTALL]: Fixed line to large.

File: src/utils/self_test/INSTALL
Comment (SABollinger) [reply src/utils/self_test/INSTALL]: Fixed line to long.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c]: Fixed,

Comment (SABollinger) [reply src/utils/self_test/self_test.c]: fixed.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c]: fixed

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c]: fixed

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c]: fixed

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1981]: fixed

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c]: fixed

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:2015]: Fixed.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1998]: ZFixed.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1944]: optind not defined in include file.  Need here.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1873]: Fixed.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1799]: Aligned.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1767]: Aligned

**Comment by SABollinger** [comment]
Waiting on code review from @mallove79 @aosinskx @jolivier23 @frostedcmos

**Comment by SABollinger** [comment]
Looking for code review from @frostedcmos  @mallove79  @aosinskx

**Review by mallove79 — COMMENTED** [review]
Seven variants of three-node self test pass in my dev env on `wolf-17vm{7,8,9,10}` (see avocado test results [here](https://github.com/daos-stack/daos/files/5812180/avocado__cart-921__selftest-three-node.tar.gz)).

File: src/utils/self_test/src/configini.c
Comment (SABollinger) [reply src/utils/self_test/src/configini.c:40]: We could, for the time being I think if we just keep to 1 type of comments is best.  Since this is new, just allow for one type.  No need to worry about back compatibility.

File: src/utils/self_test/self_test.c
Comment (SABollinger) [reply src/utils/self_test/self_test.c:1094]: The first line of the help explicitly state that the file input is read first and then any other command line input argument takes precedence.

I supplied 2 examples  for the invocation to show a simple example of using a configuration file and that of the standard method.  I can change that or rework it.  Yes, the file commands should be included in the general invocation form,

File: src/utils/self_test/etc/selftest.cnf
Comment (SABollinger) [reply src/utils/self_test/etc/selftest.cnf:1]: Added comment.

**Comment by SABollinger** [comment]
Waiting for code review from @jolivier23 @sylviachanoiyee @aosinskx
 @frostedcmos

File: Jenkinsfile
Comment (jolivier23) [inline Jenkinsfile:43]: what is this?

File: src/utils/self_test/tests/Makefile
Comment (jolivier23) [inline src/utils/self_test/tests/Makefile:1]: why are we adding makefiles ?

**Review by jolivier23 — CHANGES_REQUESTED** [review]
Please follow guidelines in 

https://wiki.hpdd.intel.com/display/DC/Commit+Comments

for the title and description of this PR

File: Jenkinsfile
Comment (SABollinger) [reply Jenkinsfile:43]: This was there to try to help debugging on centos and leap 15 system.  It was not suppose to go into source control and has been eliminated.

File: src/utils/self_test/tests/Makefile
Comment (SABollinger) [reply src/utils/self_test/tests/Makefile:1]: When Ethan brought over the configini library/object, he created a simple makefile and tests to verify that nothing was broken while doing the port.  I have kept these in as a sanity check while I was cleaning up the code (a lot to do) and adding new functions.  I kept them in for that purpose so debugging any changes to the file could be kept very local.  These are not necessary along with the utils/self_test/test directory, which is also new.  However, they could be useful if we are to expand the usage of configini.

**Comment by SABollinger** [comment]
CART 946 adds statistics verification to self_test.  It is build on top of CART 912.  At this point, I believe that  this cart should be closed and comments should be address  to  CART 946, which should be going out for review in the next day or so.

**Review by jolivier23 — COMMENTED** [review]
Can you please move this to a draft if not ready for review.  For some reason, Github processed codeowners on a bunch of old reviews

**Pull request closed without merging**
